sdk/keyvault/azure-keyvault-certificates/migration_guide.md

```diff
@@ -0,0 +1,288 @@
+
+There were several areas of consistent feedback expressed across the Azure client library ecosystem. One of the most important is that the client libraries for different Azure services have not had a consistent approach to organization, naming, and API structure. Additionally, many developers have felt that the learning curve was difficult, and the APIs did not offer a good, approachable, and consistent onboarding story for those learning Azure or exploring a specific Azure service.
+
+To try and improve the development experience across Azure services, a set of uniform [design guidelines](https://azure.github.io/azure-sdk/general_introduction.html) was created for all languages to drive a consistent experience with established API patterns for all services. A set of [Python-specific guidelines](https://azure.github.io/azure-sdk/python_introduction.html) was also introduced to ensure that Python clients have a natural and idiomatic feel with respect to the Python ecosystem. Further details are available in the guidelines for those interested.
+
+### Cross Service SDK improvements
+
+The modern Key Vault client library also provides the ability to share in some of the cross-service improvements made to the Azure development experience, such as 
+- using the new [`azure-identity`](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/identity/azure-identity/README.md) library to share a single authentication approach between clients
+- a unified logging and diagnostics pipeline offering a common view of the activities across each of the client libraries
+
+## Important changes
+
+### Separate packages and clients
+
+In the interest of simplifying the API `azure-keyvault` and `KeyVaultClient` were split into separate packages and clients:
+
+- `azure-keyvault-certificates` contains `CertificateClient` for working with certificates.
+- [`azure-keyvault-keys`](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/keyvault/azure-keyvault-keys/README.md) contains `KeyClient` for working with keys and `CryptographyClient` for performing cryptographic operations.
+- [`azure-keyvault-secrets`](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/keyvault/azure-keyvault-secrets/README.md) contains `SecretClient` for working with secrets.
+
+### Client constructors
+
+Across all modern Azure client libraries, clients consistently take an endpoint or connection string along with token credentials. This differs from `KeyVaultClient`, which took an authentication delegate and could be used for multiple Key Vault endpoints.
+
+#### Authenticating
+
+Previously in `azure-keyvault` you could create a `KeyVaultClient` by using `ServicePrincipalCredentials` from `azure-common-credentials`:
```

Comment:
This is kind of tricky. Typically, you'd import the class from `azure.common` but actually it's [defined in msrestazure](https://github.com/Azure/azure-sdk-for-python/blob/3808ad57daced04b57ff43018045b977105573ab/sdk/core/azure-common/azure/common/credentials.py#L116) and [the documentation](https://docs.microsoft.com/python/api/msrestazure/msrestazure.azure_active_directory?view=azure-python) is associated with that package. And azure-common-credentials doesn't exist. So maybe it's clearest to use the namespace here:
```suggestion
Previously in `azure-keyvault` you could create a `KeyVaultClient` by using `ServicePrincipalCredentials` from `azure.common`:
```

Reply:
This makes sense, thanks for the context! I wasn't sure about the authentication story since I never used the track 1 library, so this is really helpful